packages/primitives/src/block-quotation/index.native.js

```diff
@@ -13,20 +13,31 @@ import { withPreferredColorScheme } from '@wordpress/compose';
 import styles from './style.scss';
 
 export const BlockQuotation = withPreferredColorScheme( ( props ) => {
-	const { getStylesFromColorScheme } = props;
+	const { getStylesFromColorScheme, style } = props;
 
-	const blockQuoteStyle = getStylesFromColorScheme(
-		styles.wpBlockQuoteLight,
-		styles.wpBlockQuoteDark
-	);
+	const blockQuoteStyle = [
+		getStylesFromColorScheme(
+			styles.wpBlockQuoteLight,
+			styles.wpBlockQuoteDark
+		),
+		style?.color && {
+			borderLeftColor: style.color,
+		},
+	];
 	const newChildren = Children.map( props.children, ( child ) => {
 		if ( child && child.props.identifier === 'citation' ) {
 			return cloneElement( child, {
-				style: styles.wpBlockQuoteCitation,
+				style: {
+					...styles.wpBlockQuoteCitation,
+					...( style?.color ? { color: style.color } : {} ),
+				},
 			} );
 		}
 		if ( child && child.props.identifier === 'value' ) {
-			return cloneElement( child, { tagsToEliminate: [ 'div' ] } );
+			return cloneElement( child, {
+				tagsToEliminate: [ 'div' ],
+				...( style?.color ? { style: { color: style.color } } : {} ),
```

Comment:
Same as above, we could use the created const instead of doing that check here :)